vms/evm/warp/backend.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package warp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/libevm/log"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ava-labs/avalanchego/cache/lru"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p/acp118"
+	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
+	"github.com/ava-labs/avalanchego/vms/evm/uptimetracker"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp/payload"
+
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+)
+
+var (
+	_                         Backend = (*backend)(nil)
+	ErrValidateBlock                  = errors.New("failed to validate block message")
+	ErrVerifyWarpMessage              = errors.New("failed to verify warp message")
+	errParsingOffChainMessage         = errors.New("failed to parse off-chain message")
+
+	messageCacheSize = 500
+)
+
+type BlockClient interface {
```

Comment:
It feels weird that we call this a `BlockClient` - it implies that there is some network call going on under-the-hood. I feel like we could either go more generic and call this a `BlockStore` or a `BlockDB` or even just call this the `VM`.